metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/aop/ReleaseDateAspect.java

```diff
@@ -17,64 +17,67 @@
+ * An @Aspect for ensuring only published entities are returned through checking an entity's releaseDate field
  */
 @Aspect
-public class StudyReleaseDateAspect {
+public class ReleaseDateAspect {
 
     /**
-     * An @Around advice for StudyRepository.findAll(..) method execution.
+     * An @Around advice for repositories.*.findAll(..) method execution.
      *
-     * It takes the first argument of the join point method and adds a new Predicate for checking a study's
-     * releaseDate field. It then calls the join point method with updated arguments.
+     * Check the release field of each entity and return only published entities.
      *
      * @param proceedingJoinPoint
      * @return the return object from join point method execution
      * @throws Throwable
      */
-    @Around("execution(* uk.ac.ebi.ampt2d.metadata.persistence.repositories.StudyRepository.findAll(..))")
-    public Object filterOnReleaseDateAdviceFindAll(ProceedingJoinPoint proceedingJoinPoint) throws Throwable {
-        Object[] args = proceedingJoinPoint.getArgs();
-        Predicate predicate = (Predicate) args[0];
-
-        QStudy study = QStudy.study;
-        Predicate dateRestrictedPredicate = study.releaseDate.between(null, LocalDate.now()).and(predicate);
 
-        args[0] = dateRestrictedPredicate;
-
-        return proceedingJoinPoint.proceed(args);
+    @Around("execution(* uk.ac.ebi.ampt2d.metadata.persistence.repositories.*.findAll(..))")
+    public Object filterOnReleaseDateAdviceFindAll(ProceedingJoinPoint proceedingJoinPoint) throws Throwable {
+        Iterable<Object> results = (Iterable<Object>) proceedingJoinPoint.proceed();
+        Iterator<Object> i = results.iterator();
+        while (i.hasNext()) {
+            Object result = i.next();
+            if (processReleaseControlledObject(result) == null) {
+                i.remove();
+            }
+        }
+        return results;
     }
 
     /**
      * An @Around advice for CrudRepository.findOne(..) method execution
      *
-     * It takes the returned object from join point method execution and check the returned object. If the returned
-     * object is not a Study object, return as it is. If the returned object is a Study object, check the study's
-     * releaseDate field. Return null if the releaseDate is a date in the future.
+     * It takes the returned object from join point method execution and checks the returned object. If the returned
+     * object is not a Auditable object, return as it is. If the returned object is an Auditable object, check its
+     * release date. Return null if the release date is a date in the future.
      *
      * @param proceedingJoinPoint
      * @return null or object
      * @throws Throwable
      */
-    @Around("execution(* org.springframework.data.repository.CrudRepository.findOne(java.io.Serializable))")
+    @Around("execution(* uk.ac.ebi.ampt2d.metadata.persistence.repositories.*.findOne(java.io.Serializable))")
```

Comment:
It should be crudRepository only, if not we can't patch an unreleased study.

Reply:
You're right, thank you! Fixed in https://github.com/EBIvariation/metadata-ws/pull/53/commits/8ffa5dcca48166a32ac23b29ad55df341f95f290.